internal/pkg/otel/receivers/elasticmonitoring/receiver.go

```diff
@@ -0,0 +1,127 @@
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/otelbeat/otelmap"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"go.uber.org/zap"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/receiver"
+)
+
+type monitoringReceiver struct {
+	logger   *zap.Logger
+	config   *Config
+	consumer consumer.Logs
+
+	runCtx context.Context
+	cancel context.CancelFunc
+
+	// done is closed to signal that the receiver is finished shutting down.
+	// We use a channel rather than a wait group so the wait duration can
+	// be controlled by a provided context.
+	done chan struct{}
+}
+
+func createReceiver(
+	ctx context.Context,
+	set receiver.Settings,
+	baseCfg component.Config,
+	consumer consumer.Logs,
+) (receiver.Logs, error) {
+	cfg := baseCfg.(*Config)
+
+	runCtx, cancel := context.WithCancel(context.Background())
+
+	return &monitoringReceiver{
+		logger:   set.Logger,
+		config:   cfg,
+		consumer: consumer,
+		runCtx:   runCtx,
+		cancel:   cancel,
+		done:     make(chan struct{}),
+	}, nil
+}
+
+func (mr *monitoringReceiver) Start(ctx context.Context, _ component.Host) error {
+	go func() {
+		defer close(mr.done)
+		mr.run()
+	}()
+	return nil
+}
+
+func (mr *monitoringReceiver) Shutdown(ctx context.Context) error {
+	mr.cancel()
+	// Wait for the run loop to stop, but return immediately if the context
+	// is cancelled.
+	select {
+	case <-mr.done:
+	case <-ctx.Done():
+	}
+	return nil
+}
+
+func (mr *monitoringReceiver) run() {
+	mr.updateMetrics()
+	for mr.runCtx.Err() == nil {
+		select {
+		case <-mr.runCtx.Done():
+		case <-time.After(mr.config.Interval):
+			mr.updateMetrics()
+		}
+	}
+}
+
+func (mr *monitoringReceiver) updateMetrics() {
+	pLogs := plog.NewLogs()
+	resourceLogs := pLogs.ResourceLogs().AppendEmpty()
+	sourceLogs := resourceLogs.ScopeLogs().AppendEmpty()
+	logRecords := sourceLogs.LogRecords()
+	logRecord := logRecords.AppendEmpty()
+
+	// Initialize to the configured event template
+	beatEvent := mapstr.M(mr.config.EventTemplate.Fields).Clone()
+
+	// Add internal telemetry data
+	addMetricsFields(mr.runCtx, &beatEvent)
+
+	// Set timestamp
+	now := time.Now()
+	beatEvent["@timestamp"] = now
+	timestamp := pcommon.NewTimestampFromTime(now)
+	logRecord.SetTimestamp(timestamp)
+	logRecord.SetObservedTimestamp(timestamp)
+
+	// Convert fields to OTel-primitive types, if needed
+	otelmap.ConvertNonPrimitive(beatEvent)
+
+	// Add data_stream metadata to the log record attributes
+	if val, _ := beatEvent.GetValue("data_stream"); val != nil {
+		for _, subField := range []string{"dataset", "namespace", "type"} {
+			value, err := beatEvent.GetValue("data_stream." + subField)
+			if vStr, ok := value.(string); ok && err == nil {
+				// set log record attribute only if value is non empty
+				logRecord.Attributes().PutStr("data_stream."+subField, vStr)
+			}
+		}
+
+	}
```

Comment:
I'm pretty sure this is the same as otelconsumer.  Not something we have to do for this PR, but if it is it would be nice to consolidate this logic.